### DISCUSSION TITLE
Can I disable WHITESPACE_RULE just for Markdown?

### DISCUSSION THREAD

**Discussion by yzhang-gh** [valentjn/vscode-ltex#218] [Q&A] [answered]
Thanks for the great extension!

Just wondering whether it is possible to configure `ltex.disabledRules` (or `ltex.hiddenFalsePositives`) _specific to Markdown documents_. For example, I'd like to disable `WHITESPACE_RULE` as I usually add extra spaces in Markdown tables to make the content aligned.

**Comment by valentjn** [comment, marked as answer]
Thanks for the suggestion. Up to now, LT<sub>E</sub>X didn't treat tables specially and forwarded them directly to LanguageTool, which led to the false positives (which have annoyed me as well). Actually, it's surprising there were not more false positives with all the dashes and pipes.

It's better to solve the problem at its root, i.e., support tables instead of disabling rules for the whole document, which might lead to false negatives elsewhere.

I didn't know this, but there are lots of extensions for LT<sub>E</sub>X's Markdown parser Flexmark. [One of them](https://github.com/vsch/flexmark-java/wiki/Tables-Extension) is for [tables as in GitHub Flavored Markdown](https://github.github.com/gfm/#tables-extension-). [I enabled that](https://github.com/valentjn/ltex-ls/commit/d7a93e8a00e78bb8f5aa607b5d5a33f74d8b0ff4) and I think the false negatives about too much whitespace in tables should be gone now/in the next release of LT<sub>E</sub>X :+1:

Regarding the general question, I'm hesitating to (further) split settings for different markup languages (or enable users to apply them only for specific markup languages), as this will increase UX complexity, and when done consistently, all settings could be split up like that. Ideally, in my head, LT<sub>E</sub>X is a one-size-fits-all solution that does all the heavy lifting in the background, and users shouldn't have to worry too much about doing/configuring things differently for different markup languages.

**Reply by yzhang-gh** [reply]
> I think the false negatives about too much whitespace in tables should be gone now/in the next release of LTEX 👍

That will be great 😄.

> Regarding the general question, I'm hesitating to (further) split settings for different markup languages

I see. And I think it (extra config) won't add much value once the GFM table can be recognized.

---

BTW, I found another related issue, which is about the Markdown line break (equals to `<br>` in HTML).

```markdown
A paragraph  <!-- two trailing whitespaces here -->
still this paragraph

A paragraph<!-- GFM -->
still this paragraph
```

I got

```markdown
A paragraph  
            ^ "Possible typo: you repeated a whitespace. LTeX - WHITESPACE_RULE"
still this paragraph
^

A paragraph
          ^
still this paragraph
^
```

**Reply by valentjn** [reply]
There was a bug when handling `\r\n` line terminators (so I guess you're on Windows), which resulted in this false positive. [This is fixed now as well.](https://github.com/valentjn/ltex-ls/commit/e0b6fa6435a31bdc227fa27d79c2f1d630789688)

**Reply by yzhang-gh** [reply]
That's true 😄. Many thanks.
